lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -59,6 +60,9 @@ public class ProcessJourneyEventHandler
     private static final Logger LOGGER = LogManager.getLogger();
     private static final String CORE_SESSION_TIMEOUT_STATE = "CORE_SESSION_TIMEOUT";
     private static final String NEXT_EVENT = "next";
+    private static final String END_SESSION_EVENT = "build-client-oauth-response";
+    private static final StepResponse END_SESSION_RESPONSE =
+            new ProcessStepResponse("build-client-oauth-response", null, null);
```

Comment:
These are intentionally different - and ideally we'd change the first one to reflect that.
- `END_SESSION_EVENT` is the journey _event_ that the frontend triggers to indicate that the session should be immediately ended and returned to the RP (so perhaps could be `end-session` or similar)
- `END_SESSION_RESPONSE` is the journey _step_ that is executed in response to that event (and refers to the lambda name)

Reply:
Am I correct in thinking that the only place that can generate the `END_SESSION_EVENT` is from the `pyi-timeout-recoverable` page in the frontend?

Reply:
That's my understanding - but if you think it can happen elsewhere then let me know!

Reply:
No, all good, just getting my head around it.